book/src/proposals/leader-duplicate-block-slashing.md

```diff
@@ -0,0 +1,223 @@
+   and index, then that means there has been a duplicate transmission. If such
+   a conflicting version is detected before the block is completed, then 
+   validators drop the block. If the block is already completed, the validators
+   simply drop the duplicate and rely on the procedure outlined in the section
+   `Repairing Multiple versions of the Same Slot` below.
+
+2) `SlotMeta` column family: `(slot, blockhash)`
+Each version of a slot will have different `consumed`, `received` and 
+potentially different children as well. Thus you need a separate SlotMeta
+for each version.
+
+3) `ErasureMeta` and `IndexMeta` column families: `(slot, blockhash)`
+Each version of a slot can potentially have different erasure configurations,
+FEC blocks etc. Each of these needs to be tracked separately
+
+4) `Roots` column family: `(slot, blockhash)`
+Roots will have to know which version was rooted so we can serve the correct
+version through Repairman.
+
+5) `Dead Slots` column family: `(slot, blockhash)`
+Validators need to specify which version of a slot is dead. For instance if one
+version of a slot doesn't pass entry verification, but another version does,
+then the two versions must be distinguishable. 
+
+An issue to note here is that if some version of a slot violates correctness
+before the slot is finished, then the validator does not know what the ending
+blockhash is and thus cannot store the slot as dead. In these cases we can drop
+the entire slot and wait for repair. Because repair will include a merkle proof
+of each shred for each repair (which includes the final blockhash `B`), 
+then the advantage there is if a repaired shred fails to play, we know the
+entire version of that slot with blockhash `B` is no good, and we can store 
+`(slot, B)` in the Dead Slots column family and ignore all forks that build on
+ top of `(slot, B)`.
+
+### Repairing Multiple versions of the Same Slot
+Repair is augmented with a blockhash. The various types of repairs:
+
+`Shred(slot, blockhash, index)` - Ask for a specific shred in a slot
+with version `blockhash`. 
+
+Notes:
+
+1) Setting `blockhash` to `Hash::default()` means this validator assumes there
+is only one version of this slot in the cluster and will acccept any shred
+for `slot` and `index`. However, if a validator gets a mixed bag of such
+shreds that do not chain, they will then drop the entire `slot`, and only
+repair `slot` if some later child slot chains back to `slot` (this will be 
+initiated by `Orphan` repairs below). 
+
+2) Repair responses need to be tied to a particular blockhash. However, 
+because repair responses are limited to the size of a shred, we cannot
+include the blockhash in the repair response. This means we need to include 
+some repair "cookie" in the request + response that maps some number of bits
+to a particular `blockhash`. This breaks down if there are more versions of 
+this slot than can be tracked by the number of bits allocated to the cookie.
```

Comment:
Worried about this, also we need to shrink the shred size by the size of this cookie @pgarg66

Reply:
it'd be worthwhile to say why "repair responses need to be tied to a particular blockhash",  anti-spam?

what if repair responses were xor'd with the request cookie or the blockhash?

Reply:
Partially anti-spam (although people can still send bogus shreds/bogus cookies, hence @aeyakovenko suggested the merkle tree to figure out if it was the leader or the repair sender who sent the bogus shred), but also to figure out which off those shreds chain to which previous shreds when there are multiple versions of the same slot.

Reply:
put the rationale in the document, please?